src/proxy/momento/src/listener.rs

```diff
@@ -17,7 +19,11 @@ pub(crate) async fn listener(
         if let Ok((socket, _)) = listener.accept().await {
             TCP_ACCEPT.increment();
 
-            let client = client_builder.clone().build();
+            let client = client_builder.clone().build().unwrap_or_else(|e| {
+                // Note: this will not happen since we validated the client build in the main thread already
+                eprintln!("could not create cache client: {}", e);
+                std::process::exit(1);
+            });
```

Comment:
This `else` block is vacuous since in the main thread we already verify we can build the client.